Restructure Managing container images (CQA) (#4972)
theforeman/foreman-documentation: guides/common/modules/proc_pushing-container-images-to-project.adoc

:_mod-docs-content-type: PROCEDURE

[id="pushing-container-images-to-{project-context}"]
= Pushing container images to {Project}

[role="_abstract"]

> Push container images to the {Project} container registry to create repositories and distribute custom content by using Podman or Docker.

Reviewer: Maybe split this into two parts because at first glance, I though "distribute custom content" was done by Podman/Docker.

```suggestion
Push container images to the {Project} container registry to distribute custom container images.
You can use Podman or Docker to consume container images on hosts.
```

alternative: shorten "custom container images." to "cstom content.".

Reviewer 2: +1, agree that the wording made me think podman was doing the distribution.

.Prerequisites
* Review xref:considerations-for-using-the-{project-context}-container-registry[].
* To push content to {Project}, ensure your {Project} account has the `edit_products` permission.
* Ensure that a product exists before pushing a repository.
For more information, see xref:creating-a-custom-product-by-using-web-ui[].
* Your {Project} account has a role that grants the `create_personal_access_tokens` permission to generate an access token for authentication.

> The token generates automatically when you run `podman login`.

Reviewer: same as above.

Author: ```suggestion
{Project} automatically generates a token when you run `podman login`.
```

.Procedure
. Log in to the container registry:
+
[options="nowrap", subs="+quotes,attributes"]
----

> # podman login _{foreman-example-com}_

Reviewer: ```suggestion
$ podman login _{foreman-example-com}_
```

Unless there's a reason that only `root` can do this?

Reviewer 3: Nope, any user can do that.

Author: So this change applies to all podman commands perhaps?

Reviewer: Almost all. I guess all but the ones that use an authfile in `/etc/foreman/`, i.e.

````
$ rg authfile
guides/common/modules/proc_upgrading-your-project-server.adoc
32:# podman login --authfile /etc/foreman/registry-auth.json registry.redhat.io

guides/common/modules/proc_installing-insights-iop-on-a-connected-project-context-server.adoc
35:# podman login --authfile /etc/foreman/registry-auth.json registry.redhat.io
````

looks correct to me.

Author: I meant just in this assembly :) not changing anything else.

----
. Push a container image to the {Project} container registry.
To indicate which organization, product, and repository the container image belongs to, include the organization and product in the container registry name.
You can address the container destination by using one of the following schemas:
+
[options="nowrap", subs="+quotes,attributes"]
----

> $ podman push _My_Container_Image_Hash_ _{foreman-example-com}_/_My_Organization_Label_/_My_Product_Label_/_My_Repository_Name_[:__My_Tag__]

Reviewer 2: Since there is another document that references label-based and ID-based schemas, do you think it's worthwhile here to call them out specifically? A user may try grepping around for "ID-based schema" and come up short.

Author: Done.
